engine/schema/src/com/cloud/host/dao/HostDaoImpl.java

```diff
@@ -1087,4 +1087,11 @@ public HostVO findByPublicIp(String publicIp) {
         sc.addAnd("dataCenterId", SearchCriteria.Op.EQ, zoneId);
         return customSearch(sc, null);
     }
+
+    @Override
+    public List<HostVO> listByType(Host.Type type) {
+        SearchCriteria<HostVO> sc = TypeSearch.create();
+        sc.setParameters("type", type);
+        return listBy(sc);
```

Comment:
@rhtyd 
listBy(sc) in GenericDao base returns an empty ArrayList if there is no matching data. 
listByType doesn't return null